src/Model/Types.ts

```diff
@@ -0,0 +1,65 @@
+/**
+ * @Author: Rostislav Simonik <rostislav.simonik>
+ * @Date:   2018-07-17T03:01:48+02:00
+ * @Email:  rostislav.simonik@technologystudio.sk
+ * @Copyright: Technology Studio
+**/
+
+import type {
+  NavigationAction,
+  Router,
+  RouterConfigOptions,
+} from '@react-navigation/native'
+import type UseOnActionType from '@react-navigation/core/lib/typescript/src/useOnAction'
+import type { NavigationState } from '@react-navigation/routers'
+import type { ValuesType } from 'utility-types'
+
+export type ConditionalNavigationState = {
+  condition: Condition,
+  logicalTimestamp: number,
+  postponedAction: NavigationAction | null,
+  previousState: NavigationState,
+}
+
+export type Condition = {
+  key: string,
+}
+
+export const navigatorTypes = {
```

Comment:
can be an enum ?

Reply:
changed to enum and removed `utility-types` package, it was used only here